libbeat/processors/add_host_geo/add_host_geo.go

```diff
@@ -0,0 +1,59 @@
+package add_geo
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+func init() {
+	processors.RegisterPlugin("add_host_geo", newAddGeo)
+}
+
+func newAddGeo(c *common.Config) (processors.Processor, error) {
+	config := &Config{}
+	c.Unpack(config)
+
+	ag := addGeo{
+		"name":             config.Name,
+		"location":         config.Location,
+		"continent_name":   config.ContinentName,
+		"country_iso_code": config.CountryISOCode,
+		"region_name":      config.RegionName,
+		"city_name":        config.CityName,
+	}
+	// Delete any empty values
+	for k, v := range ag {
+		if v == "" {
+			delete(ag, k)
+		}
+	}
+
+	return ag, nil
+}
+
+type Config struct {
+	Name           string `config:"name"`
+	Location       string `config:"location"`
+	ContinentName  string `config:"continent_name"`
+	CountryISOCode string `config:"country_iso_code"`
+	RegionName     string `config:"region_name"`
+	CityName       string `config:"city_name"`
+}
```

Comment:
I'd prefer to have validation for at least the fields that are stored as geo point in Elasticsearch here.

Reply:
Ah, so we'd check the format to be a lat lon pair? I think that makes sense. Anything beyond that?

Reply:
yes, at least lat-lon pair.

The others are kind free form text and validating these might require a location database, right? We even store the others as keyword, so not that important. Validating them might be too much effort here (would be nice if we can do some simple validation - regex like -, but no hard requirement for me).

Most important is that indexing won't fail due to misconfiguration of the processor + maps in kibana dashboards will work correctly, cause lat lon pairs are correct and make sense.

Reply:
Validation added :)